cmake/CubCudaConfig.cmake

```diff
@@ -1,87 +1,23 @@
 enable_language(CUDA)
 
-if (NOT CUB_IN_THRUST)
-  message(FATAL_ERROR
-    "Building CUB as a standalone project is no longer supported. "
-    "Use the Thrust repo instead.")
-endif()
-
-set(CUB_CUDA_FLAGS_BASE "${THRUST_CUDA_FLAGS_BASE}")
-set(CUB_CUDA_FLAGS_RDC "${THRUST_CUDA_FLAGS_RDC}")
-set(CUB_CUDA_FLAGS_NO_RDC "${THRUST_CUDA_FLAGS_NO_RDC}")
-
-# Update the enabled architectures list from thrust
-foreach (arch IN LISTS THRUST_KNOWN_COMPUTE_ARCHS)
-  if (THRUST_ENABLE_COMPUTE_${arch})
-    set(CUB_ENABLE_COMPUTE_${arch} True)
-    string(APPEND arch_message " sm_${arch}")
-  else()
-    set(CUB_ENABLE_COMPUTE_${arch} False)
-  endif()
-endforeach()
-
-message(STATUS ${arch_message})
-
 #
-# RDC options:
+# Architecture options:
 #
 
-# RDC is off by default in NVCC and on by default in NVC++. Turning off RDC
-# isn't currently supported by NVC++. So, we default to RDC off for NVCC and
-# RDC on for NVC++.
-set(option_init OFF)
-if ("NVCXX" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
-  set(option_init ON)
-endif()
-
-option(CUB_ENABLE_TESTS_WITH_RDC
-  "Build all CUB tests with RDC; tests that require RDC are not affected by this option."
-  ${option_init}
-)
-
-option(CUB_ENABLE_EXAMPLES_WITH_RDC
-  "Build all CUB examples with RDC; examples which require RDC are not affected by this option."
-  ${option_init}
-)
+# Create a new arch list that only contains arches that support CDP:
+set(CUB_CUDA_ARCHITECTURES ${CMAKE_CUDA_ARCHITECTURES})
+set(CUB_CUDA_ARCHITECTURES_RDC ${CUB_CUDA_ARCHITECTURES})
+list(FILTER CUB_CUDA_ARCHITECTURES_RDC EXCLUDE REGEX "53|62|72|90")
```

Comment:
This will break for `native|all|all-major` under certain conditions.

Reply:
For now, we'll emit an error advising the user to explicitly specify arches or turn off `ENABLE_RDC` options.

We can also do introspection ourselves at some point to determine this if needed.